R/utilities.R

```diff
@@ -2741,45 +2849,52 @@ collate_lymphgen = function(sample_table,
 #' @import tidyverse
 #'
 #' @examples
-#' sample_table <- data.frame (sample_id  = c("09-33003T", "05-18426T", "BLGSP-71-06-00174-01A-01D", "BLGSP-71-22-00347-01A-12E", "SP59282"))
-#' qc_metrics = collate_qc_results(sample_table = sample_table, seq_type_filter = "genome")
+#' qc_metrics = collate_qc_results(sample_table = sample_table)
 #'
 collate_qc_results = function(sample_table,
-                              seq_type_filter = "genome"){
-
+                              seq_type_filter){
+  
   if(! seq_type_filter %in% c("genome", "capture")){
     stop("Please provide a valid seq_type (\"genome\" or \"capture\").")
   }
-
+  
   #get paths
   base = config::get("project_base") #base
   qc_template = config::get("qc_met") #qc metric template
   qc_path = glue::glue(qc_template) #glue wildcards
   icgc_qc_path_full = paste0(base, qc_path) #combine for icgc_dart qc metric path
   gambl_qc = gsub("icgc_dart", "gambl", qc_path) #use gsub to substitute icgc_dart in qc path with gambl
   gambl_qc_path_full = paste0(base, gambl_qc) #combine for gambl qc metric path
-
+  
   #read in icgc qc data, rename sample id column and filter on samples in sample id in sample_table
   icgc_qc = suppressMessages(read_tsv(icgc_qc_path_full)) %>%
-    rename(sample_id = UID) %>%
-    dplyr::filter(sample_id %in% dplyr::pull(sample_table, sample_id))
-
+      dplyr::rename(sample_id = UID) %>%
+      dplyr::select(-SeqType)
+  
   #read in gambl qc data (if seq_type_filter set to "genome"), rename sample id column and filter on samples in sample id in sample_table
   if(seq_type_filter == "genome"){
     gambl_qc = suppressMessages(read_tsv(gambl_qc_path_full)) %>%
-      rename(sample_id = UID) %>%
-      dplyr::filter(sample_id %in% dplyr::pull(sample_table, sample_id))
-
+      dplyr::rename(sample_id = UID) %>%
+      dplyr::select(-SeqType)
+    
     #join gambl and icgc QC data
     full_qc = rbind(gambl_qc, icgc_qc)
+    sample_table = left_join(sample_table, full_qc)
 
-    return(full_qc)
+    #print n samples with QC metrics
+    qc_samples = length(unique(full_qc$sample_id))
+    message(paste("QC metrics for", qc_samples, "samples retrieved"))
+    
+  }else{
+    message("Currently, seq_type_filter = \"capture\" is only available for unix_group \"icgc_dart\". Only QC metrics for icgc_dart will be returned.")
```

Comment:
I'm confused. This function is implying that capture isn't supported for the other unix group but why?

Reply:
When this function was added, we didn't have any QC metrics generated for capture samples with UNIX group gambl. If I remember correctly. Kostia told me this will be added at some point. Thus the function can not retrieve QC data for gambl capture samples, the file does not exists (yet?).

Reply:
What does this function return when you provide the capture metadata as the samples table and run with seq_type_filter = "capture"?

Reply:
```
#metadata
> capture_table = get_gambl_metadata(seq_type_filter = "capture") %>% 
    dplyr::select(sample_id, patient_id, biopsy_id, seq_type)
                                                                                                                                                                                                                                                          
#qc
> capture_results = collate_qc_results(sample_table = capture_table, seq_type_filter = "capture")
Currently, seq_type_filter = "capture" is only available for unix_group "icgc_dart". Only QC metrics for icgc_dart will be returned.                                                                                                                      
Joining, by = "sample_id"
QC metrics for 124 samples retrieved

> head(capture_results)
# A tibble: 6 × 18
  sample_id       patient_id biopsy_id  seq_type AverageBaseQuality AverageInsertSize AverageReadLength PairsOnDiffCHR TotalReads TotalUniquelyMap… TotalUnmappedre… TotalDuplicated… ProportionReads… ProportionReads…
  <chr>           <chr>      <chr>      <chr>                 <dbl>             <dbl>             <dbl>          <dbl>      <dbl>             <dbl>            <dbl>            <dbl>            <dbl>            <dbl>
1 02-11616_tumorA 02-11616   S02-1811   capture                  NA                NA                NA             NA         NA                NA               NA               NA               NA               NA
2 02-17305_tumorB 02-17305   VS08-18702 capture                  NA                NA                NA             NA         NA                NA               NA               NA               NA               NA
3 03-19969_tumorB 03-19969   S04-12701  capture                  NA                NA                NA             NA         NA                NA               NA               NA               NA               NA
4 03-23488_tumorA 03-23488   SU03-8870  capture                  NA                NA                NA             NA         NA                NA               NA               NA               NA               NA
5 04-11038_tumorA 04-11038   VS04-661   capture                  NA                NA                NA             NA         NA                NA               NA               NA               NA               NA
6 04-11038_tumorB 04-11038   BS18-1998  capture                  NA                NA                NA             NA         NA                NA               NA               NA               NA               NA
# … with 4 more variables: MeanCorrectedCoverage <dbl>, ProportionTargetsNoCoverage <dbl>, ProportionCoverage10x <dbl>, ProportionCoverage30x <dbl>
```

Reply:
Just to show that there actually is QC data for (some of) the samples
```
> capture_results[order(capture_results$AverageBaseQuality),]
# A tibble: 2,271 × 18
   sample_id patient_id biopsy_id seq_type AverageBaseQuality AverageInsertSize AverageReadLength PairsOnDiffCHR TotalReads TotalUniquelyMapped TotalUnmappedreads TotalDuplicatedre… ProportionReads… ProportionReads…
   <chr>     <chr>      <chr>     <chr>                 <dbl>             <dbl>             <dbl>          <dbl>      <dbl>               <dbl>              <dbl>              <dbl>            <dbl>            <dbl>
 1 PBL062T   PBL062     PBL062    capture                33.8              97.9               150        1028972   56373556            56219131             154425           34289904             0.61                1
 2 PBL122T   PBL122     PBL122    capture                34.2             331.                150        4852012   66730816            66524931             205885           42018179             0.63                1
 3 PBL118T   PBL118     PBL118    capture                34.3             198.                150        2495055   84819956            84590864             229092           25782507             0.3                 1
 4 PBL068T   PBL068     PBL068    capture                34.5             196.                150        2891320  124278278           124124434             153844           62118572             0.5                 1
 5 PBL106T   PBL106     PBL106    capture                34.5             145.                150        1709854   91945686            91769605             176081           56366033             0.61                1
 6 PBL112T   PBL112     PBL112    capture                34.6             145.                150        1248420   52293838            52204803              89035           25313779             0.48                1
 7 PBL058T   PBL058     PBL058    capture                34.6             272.                150        3840289   81318044            81040974             277070           20259624             0.25                1
 8 PBL021T   PBL021     PBL021    capture                34.6             161.                150        3510318   85804928            85383321             421607           25233184             0.29                1
 9 PBL042T   PBL042     PBL042    capture                34.6             174.                150        2951602   85557642            85302995             254647           21440046             0.25                1
10 PBL123T   PBL123     PBL123    capture                34.7             138.                150         917569   34191782            34027500             164282            7281258             0.21                1
# … with 2,261 more rows, and 4 more variables: MeanCorrectedCoverage <dbl>, ProportionTargetsNoCoverage <dbl>, ProportionCoverage10x <dbl>, ProportionCoverage30x <dbl>
```